extractor/filesystem/language/java/gradlelockfile/gradlelockfile.go

```diff
@@ -107,7 +107,7 @@ func (e Extractor) Extract(ctx context.Context, input *filesystem.ScanInput) ([]
 	}
 
 	if err := scanner.Err(); err != nil {
-		return []*extractor.Inventory{}, fmt.Errorf("failed to read: %w", err)
+		return nil, fmt.Errorf("failed to read: %w", err)
```

Comment:
technically we could return `pkgs` here as per supporting "PartiallySucceeded", but I feel that's a separate concern to this PR so have just gone with `nil`